zkevm-circuits/src/evm_circuit/util/constraint_builder.rs

```diff
@@ -1024,19 +1042,29 @@ impl<'a, F: FieldExt> ConstraintBuilder<'a, F> {
     }
 
     pub(crate) fn add_constraints(&mut self, constraints: Vec<(&'static str, Expression<F>)>) {
-        self.cb.add_constraints(constraints);
+        for (name, constraint) in constraints {
+            self.add_constraint(name, constraint);
+        }
     }
 
     pub(crate) fn add_constraint(&mut self, name: &'static str, constraint: Expression<F>) {
-        self.cb.add_constraint(name, constraint);
+        let constraint = match &self.condition {
+            Some(condition) => condition.clone() * constraint,
+            None => constraint,
+        };
+
+        let constraint = self.split_expression(name, constraint, MAX_DEGREE - 2);
```

Comment:
It seems we subtract the implicit degree twice, just tested locally and the `extended_k` still remains `3` (`validate_degree` also needs to be updated).
```suggestion
        let constraint = self.split_expression(name, constraint, MAX_DEGREE);
```